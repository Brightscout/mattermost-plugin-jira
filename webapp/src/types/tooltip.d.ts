type TicketDetails = {
    assigneeName: string;
    assigneeAvatar: any;
    labels: any;
    description: string;
    summary: any;
    ticketId: any;
    jiraIcon: any;
    versions: any;
    statusKey: any;
    issueIcon: any;
}

type TicketData = {
    key: string;
    fields: TicketDataFields;
}

type TicketDataFields = {
    labels: string[];
    description: string;
    summary: string;
    project: any;
    versions: string[];
    status: {
        name: string;
    };
    issuetype: {
        iconUrl: string;
    };
}
