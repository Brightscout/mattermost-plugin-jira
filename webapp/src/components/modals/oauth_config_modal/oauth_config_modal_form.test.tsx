// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import {InstanceType} from 'types/model';

import OAuthConfigModalForm from './oauth_config_modal_form';

describe('components/OAuthConfigModalForm', () => {
    const baseActions = {
        closeModal: jest.fn().mockResolvedValue({}),
        configure: jest.fn().mockResolvedValue({}),
        addValidate: jest.fn(),
        removeValidate: jest.fn(),
        installedInstances: [
            {
                instance_id: 'https://something.atlassian.net',
                type: InstanceType.CLOUD_OAUTH,
            },
        ],
    };

    const baseProps = {
        ...baseActions,
        visible: true,
        theme: {},
    };

    test('should match snapshot', () => {
        const props = {...baseProps};
        const wrapper = shallow<OAuthConfigModalForm>(
            <OAuthConfigModalForm {...props}/>
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('should configure on submit', async () => {
        const closeModal = jest.fn().mockResolvedValue({});
        const configure = jest.fn().mockResolvedValue({});
        const props = {
            ...baseProps,
            closeModal,
            configure,
        };
        const wrapper = shallow<OAuthConfigModalForm>(
            <OAuthConfigModalForm {...props}/>
        );

        wrapper.instance().handleInstanceChange('', 'http://localhost:8080');
        expect(wrapper.state().instanceUrl).toEqual('http://localhost:8080');
        expect(wrapper.state().error).toEqual('');

        wrapper.instance().handleClientIdChange('', 'someclientid');
        expect(wrapper.state().clientId).toEqual('someclientid');
        expect(wrapper.state().error).toEqual('');

        wrapper.instance().handleClientSecretChange('', 'someclientsecret');
        expect(wrapper.state().clientSecret).toEqual('someclientsecret');
        expect(wrapper.state().error).toEqual('');
    });

    test('should show error when user is already connected to instance', async () => {
        const props = {...baseProps};
        const wrapper = shallow<OAuthConfigModalForm>(
            <OAuthConfigModalForm {...props}/>
        );

        wrapper.instance().handleInstanceChange('', 'https://something.atlassian.net');
        expect(wrapper.state().instanceUrl).toEqual('https://something.atlassian.net');
        expect(wrapper.state().error).toEqual('You have already installed this Jira instance.');
    });
});
