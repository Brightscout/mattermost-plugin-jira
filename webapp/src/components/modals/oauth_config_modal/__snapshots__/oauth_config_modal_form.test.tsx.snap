// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/OAuthConfigModalForm should match snapshot 1`] = `
<form
  onSubmit={[Function]}
  role="form"
>
  <ModalBody
    bsClass="modal-body"
    componentClass="div"
    style={
      Object {
        "backgroundColor": undefined,
        "color": undefined,
        "padding": "2em 2em 3em",
      }
    }
  >
    <Input
      addValidate={[Function]}
      disabled={false}
      helpText="Enter a Jira Cloud URL (typically, \`https://yourorg.atlassian.net\`), or just the organization part, \`yourorg\`"
      label="Jira Cloud organization"
      maxLength={null}
      onChange={[Function]}
      placeholder="https://yourorg.atlassian.net"
      readOnly={false}
      removeValidate={[Function]}
      required={true}
      type="input"
      value=""
    />
    <Input
      addValidate={[Function]}
      disabled={false}
      helpText="The client ID for the OAuth app registered with Jira"
      label="Jira OAuth Client ID"
      maxLength={null}
      onChange={[Function]}
      readOnly={false}
      removeValidate={[Function]}
      required={true}
      type="input"
      value=""
    />
    <Input
      addValidate={[Function]}
      disabled={false}
      helpText="The client secret for the OAuth app registered with Jira"
      label="Jira OAuth Client Secret"
      maxLength={null}
      onChange={[Function]}
      readOnly={false}
      removeValidate={[Function]}
      required={true}
      type="input"
      value=""
    />
  </ModalBody>
  <ModalFooter
    bsClass="modal-footer"
    componentClass="div"
    style={
      Object {
        "padding": "2rem 15px",
      }
    }
  >
    <FormButton
      btnClass="btn-link"
      defaultMessage="Cancel"
      disabled={false}
      extraClasses=""
      onClick={[Function]}
      savingMessage="Creating"
      type="button"
    />
    <FormButton
      btnClass="btn btn-primary"
      defaultMessage="Configure"
      disabled={false}
      extraClasses=""
      saving={false}
      savingMessage="Creating"
      type="submit"
    />
    <p
      className="error-text"
    />
  </ModalFooter>
</form>
`;
